src/command/cmd_funcs.c

```diff
@@ -9818,3 +9855,912 @@ cmd_mood(ProfWin* window, const char* const command, gchar** args)
+        if (value) {
+            element->note = strdup(value);
+        }
+    } else if (g_strcmp0(type, "url") == 0) {
+        element->type = VCARD_URL;
+        if (value) {
+            element->url = strdup(value);
+        }
+    } else {
+        cons_bad_cmd_usage(command);
+        free(element);
+        return TRUE;
+    }
+
+    vcard_user_add_element(element);
+    vcardwin_update();
+    return TRUE;
+}
+
+gboolean
+cmd_vcard_remove(ProfWin* window, const char* const command, gchar** args)
+{
+    if (connection_get_status() != JABBER_CONNECTED) {
+        cons_show("You are not currently connected.");
+        return TRUE;
+    }
+
+    if (args[1]) {
+        vcard_user_remove_element(atoi(args[1]));
+        cons_show("Removed element at index %d", atoi(args[1]));
+        vcardwin_update();
+    } else {
+        cons_bad_cmd_usage(command);
+    }
+    return TRUE;
+}
+
+gboolean
+cmd_vcard_get(ProfWin* window, const char* const command, gchar** args)
+{
+    char* user = args[1];
+    xmpp_ctx_t* const ctx = connection_get_ctx();
+
+    if (connection_get_status() != JABBER_CONNECTED) {
+        cons_show("You are not currently connected.");
+        return TRUE;
+    }
+
+    if (user) {
+        // get the JID when in MUC window
+        if (window->type == WIN_MUC) {
+            ProfMucWin* mucwin = (ProfMucWin*)window;
+            assert(mucwin->memcheck == PROFMUCWIN_MEMCHECK);
+
+            if (muc_anonymity_type(mucwin->roomjid) == MUC_ANONYMITY_TYPE_NONANONYMOUS) {
+                // non-anon muc: get the user's jid and send vcard request to them
+                Occupant* occupant = muc_roster_item(mucwin->roomjid, user);
+                Jid* jid_occupant = jid_create(occupant->jid);
+
+                vcard_print(ctx, window, jid_occupant->barejid);
+                jid_destroy(jid_occupant);
+            } else {
+                // anon muc: send the vcard request through the MUC's server
+                GString* full_jid = g_string_new(mucwin->roomjid);
+                g_string_append(full_jid, "/");
+                g_string_append(full_jid, user);
+
+                vcard_print(ctx, window, full_jid->str);
+
+                g_string_free(full_jid, TRUE);
+            }
+        } else {
+            char* jid = roster_barejid_from_name(user);
+            if (!jid) {
+                cons_bad_cmd_usage(command);
+                return TRUE;
+            }
+
+            vcard_print(ctx, window, jid);
+        }
+    } else {
+        if (window->type == WIN_CHAT) {
+            ProfChatWin* chatwin = (ProfChatWin*)window;
+            assert(chatwin->memcheck == PROFCHATWIN_MEMCHECK);
+
+            vcard_print(ctx, window, chatwin->barejid);
+        } else {
+            vcard_print(ctx, window, NULL);
+        }
+    }
+
+    return TRUE;
+}
+
+gboolean
+cmd_vcard_photo(ProfWin* window, const char* const command, gchar** args)
+{
+    char* operation = args[1];
+    char* user = args[2];
```

Comment:
How do we design the command to be able to get the user's own avatar
Commands like `/vcard photo open` may work but `/vcard photo open 4` (as in index) may conflict with the nick/contact parameter
Or `/vcard photo save` but `/vcard photo save filename test.png` or `/vcard photo save index 4` may lead the command into thinking that it's supposed to get an avatar from a user nicknamed `filename` or `index`

Reply:
I'm missing some context.
What exactly is `/vcard photo open 4` supposed to do?

Reply:
> I'm missing some context. What exactly is `/vcard photo open 4` supposed to do?

Download the avatar from a user nicknamed `4`, In some cases, this might get confused with downloading the 4th avatar from your vCard

Reply:
But is there a way (or should there be a way) to download multiple avatars? If yes we could just have another parameter `/vcard photo open-index 4`. We have problems like this with other commands also we would need to check how we solved things there. Either by having a new (sub)command, a command with `-` or reorganizing some commands.

Reply:
Or maybe we could have `/vcard photo open-self [<index>]` and `/vcard photo save-self [filename <filename>] [index <index>]`

Reply:
> Or maybe we could have /vcard photo open-self [<index>] and /vcard photo save-self [filename <filename>] [index <index>]

Sounds good. And for other users?

Reply:
> > Or maybe we could have /vcard photo open-self [] and /vcard photo save-self [filename ] [index ]
> 
> Sounds good. And for other users?

`/vcard photo open <nick/contact> [<index>]`
`/vcard photo save <nick/contact> [filename <filename>] [index <index>]`

Reply:
Sounds good to me.